src/netbase.cpp

```diff
@@ -214,6 +218,28 @@ CService LookupNumeric(const std::string& name, uint16_t portDefault, DNSLookupF
     return Lookup(name, portDefault, /*fAllowLookup=*/false, dns_lookup_function).value_or(CService{});
 }
 
+bool IsUnixSocketPath(const std::string& name)
+{
+#if HAVE_SOCKADDR_UN
+    if (name.find(ADDR_PREFIX_UNIX) != 0) return false;
+
+    // Split off "unix:" prefix
+    std::string str{name.substr(ADDR_PREFIX_UNIX.length())};
+
+    // Path size limit is platform-dependent
+    // see https://manpages.ubuntu.com/manpages/xenial/en/man7/unix.7.html
+    if (str.size() + 1 > sizeof(((sockaddr_un*)nullptr)->sun_path)) return false;
+
+    // Path must exist
+    fs::path path = fs::PathFromString(str);
+    if (!fs::exists(path)) return false;
```

Comment:
I wonder if this check is desirable. It makes `IsUnixSocketPath()` volatile - for example if the server is not yet started it will be false and the moment it is started it will become true, or if the server is restarted it will flip from true to false and back.

With this check `bitcoind` will refuse to start if the Tor server is not started yet. This differs from the TCP case where the Tor server can start later.

With TCP we only check if the given address is something we can _try_ to connect to. Like "ok, it is 4 digits separated by 3 dots, maybe I can connect to this later when needed". `CService::IsValid()` does not check whether somebody is listening on that address and port and that we can connect to it. An equivalent check for UNIX sockets would be "ok, if it starts with unix: and is short enough maybe I can connect to this later when needed".